sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -55,10 +56,19 @@ object DateTimeUtils {
   // this is year -17999, calculation: 50 * daysIn400Year
   final val YearZero = -17999
   final val toYearZero = to2001 + 7304850
-  final val TimeZoneGMT = TimeZone.getTimeZone("GMT")
 
   @transient lazy val defaultTimeZone = TimeZone.getDefault
 
+  // Reuse the TimeZone object as it is expensive to create in each method call.
+  final val timeZones = new ConcurrentHashMap[String, TimeZone]
```

Comment:
Actually, we only need to change all `getTimeZone(String ID)` to `getTimeZone(String ID, boolean fallback)`, (use true as fallback) to workaround the synchronized tag here.